### PR TITLE
Bump @metamask/providers to v13.0.0

### PR DESCRIPTION
**Pull request by jiexi**
Driven by a need to get rid of `window.ethereum.networkVersion` as a part of removing networkId work. This PR bumps `@metamask/providers` from v11.1.0 to v13.0.0, which introduces a deprecation warning when accessing `window.ethereum.networkVersion` (as well as `chainId` and `selectedAddress`). The goal is to get this into the wild, alert devs, and to finally remove these properties in the future.

~~Additionally, this PR bumps `browserify` from v16.5.1 to v17.0.0 which is required for polyfilling `streams.pipeline` which is now used in `@metamask/providers`~~ Completed in a separate PR already

https://github.com/MetaMask/providers/compare/v11.1.0...v13.0.0

* Fixes: https://github.com/MetaMask/MetaMask-planning/issues/1068
* See: https://github.com/MetaMask/metamask-extension/pull/20652

### Before
Screenshot 2023-09-19 at 9 36 46 AM

### After
Screenshot 2023-09-19 at 9 31 30 AM

## Manual Testing Steps
1. Open browser console in any page
2. Enter `window.ethereum.chainId`
3. You should get the value for the current selected chain ID and a deprecation warning
4. Enter `window.ethereum.chainId` again
5. You should get the value for the current selected chain ID, but no deprecation warning this time
6. Repeat 2-6 for `window.ethereum.networkVersion` and `window.ethereum.selectedAddress`

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [x] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [x] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Comment by jiexi** [comment]
@metamaskbot update-policies

**Comment by jiexi** [comment]
@metamaskbot update-policies

**Comment by adonesky1** [comment]
Can we only have one version of `browserify` in our dependency graph? Otherwise shouldn't `@metamask/providers` bring the different version requirement with it?

**Review by adonesky1 — DISMISSED** [review]
LGTM!

**Comment by jiexi** [comment]
> One thought on making this upgrades as safe and smooth as we can: If browserify@16->17 can be broken out separately and merged prior to upgrading providers to 13, that would be preferred

Resolved by https://github.com/MetaMask/metamask-extension/pull/20946 . Thank you for taking care of that!

> [@metamask/keyring-api@0.2.5](https://github.com/MetaMask/keyring-api/compare/v0.2.4...v0.2.5#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519)

Can bump this in a separate PR

> https://github.com/MetaMask/snaps/pull/1748 

Looks like this may require some manual attention. I'll try to take a look today.

> But yes, if this actually relies on a polyfill that browserify pulls in, sounds like something that should be done wither by providers or directly in metamask-extension itself?

Hard to say. It's awkward because the provider is almost always used in the browser context, but not strictly. Because it's totally valid to be used in nodejs, I think this is a concern of metamask-extension and not providers.

**Comment by jiexi** [comment]
Current provider alignment status. Still have v11 floating around

* bump provider to v13 in `@metamask/snaps-execution-environments`
  * consume in `@metamask/snaps-controllers`
  * consume in `@metamask/eth-snap-keyring`
  * consume in `@metamask/keyring-api`
    * consume in `@metamask/eth-snap-keyring`
 
bump provider to v13 in `@metamask/snaps-execution-environments`: https://github.com/MetaMask/snaps/pull/1748

File: lavamoat/build-system/policy.json [hunk 1]
```diff
@@ -1136,6 +1136,21 @@
         "string.prototype.matchall>side-channel": true
```
Comment (adonesky1) [inline lavamoat/build-system/policy.json:1136]: pretty sure we don't want to commit these changes? I believe these are OS specific.

**Comment by jiexi** [comment]
@metamaskbot update-policies

File: lavamoat/build-system/policy.json (hunk 1 above)
Comment (jiexi) [reply lavamoat/build-system/policy.json:1136]: i believe this comes from the auto generated lavamoat policies

**Comment by jiexi** [comment]
No longer blocked by having to align snaps (provider v11) and this change

**Review by adonesky1 — APPROVED** [review]
LGTM

**Review by brad-decker — APPROVED** [review]

**Review by shanejonas — APPROVED** [review]

**Pull request merged**